VungleAdapter/src/main/java/com/chartboost/mediation/vungleadapter/VungleAdapter.kt

```diff
@@ -664,153 +524,228 @@ class VungleAdapter : PartnerAdapter {
                 }
             }
 
-            Vungle.loadAd(
-                request.partnerPlacement,
-                adm,
-                adConfig,
-                object : LoadAdCallback {
-                    override fun onAdLoad(id: String) {
-                        PartnerLogController.log(LOAD_SUCCEEDED)
-                        resumeOnce(
-                            Result.success(createPartnerAd(null, request))
-                        )
-                    }
+            fun loadVungleFullScreenAd(fullscreenAd: () -> BaseFullscreenAd) {
+                fullscreenAd().apply {
+                    adListener = createFullScreenAdListener(request, listener, continuation)
+                    load(adm)
+                }
+            }
 
-                    override fun onError(placementId: String?, exception: VungleException?) {
-                        PartnerLogController.log(
-                            LOAD_FAILED,
-                            "Placement: $placementId. Error code: ${exception?.exceptionCode}. " +
-                                    "Message: ${exception?.message}"
-                        )
+            when (request.format) {
+                AdFormat.INTERSTITIAL -> loadVungleFullScreenAd {
+                    InterstitialAd(context, request.partnerPlacement, adConfig)
+                }
+
+                AdFormat.REWARDED -> loadVungleFullScreenAd {
+                    RewardedAd(context, request.partnerPlacement, adConfig)
+                }
+
+                else -> {
+                    if (request.format.key == "rewarded_interstitial") {
+                        loadVungleFullScreenAd {
+                            RewardedAd(context, request.partnerPlacement, adConfig)
+                        }
+                    } else {
+                        PartnerLogController.log(LOAD_FAILED)
                         resumeOnce(
                             Result.failure(
                                 ChartboostMediationAdException(
-                                    getChartboostMediationError(
-                                        exception
-                                    )
+                                    ChartboostMediationError.CM_LOAD_FAILURE_UNSUPPORTED_AD_FORMAT
                                 )
                             )
                         )
                     }
                 }
-            )
+            }
         }
     }
 
+
     /**
      * Attempt to show a Vungle fullscreen ad.
      *
      * @param partnerAd The [PartnerAd] object containing the Vungle ad to be shown.
-     * @param listener A [PartnerAdListener] to notify Chartboost Mediation of ad events.
+     *
+     * @return Result.success(PartnerAd) if the ad was successfully shown, Result.failure(Exception) otherwise.
      */
     private suspend fun showFullscreenAd(
-        partnerAd: PartnerAd,
-        listener: PartnerAdListener?
+        partnerAd: PartnerAd
```

Comment:
Nit: keep the comma so this shows up as a 1-line diff rather than as a 2-line diff.

Reply:
Is there a way to add this our IDEs? 😲

Reply:
Done. 👍🏼